h2/src/main/org/h2/value/DataType.java

```diff
@@ -1335,19 +1328,19 @@ public static Object getDefaultForPrimitiveType(Class<?> clazz) {
         if (clazz == Boolean.TYPE) {
             return Boolean.FALSE;
         } else if (clazz == Byte.TYPE) {
-            return Byte.valueOf((byte) 0);
+            return (byte) 0;
         } else if (clazz == Character.TYPE) {
-            return Character.valueOf((char) 0);
+            return (char) 0;
         } else if (clazz == Short.TYPE) {
-            return Short.valueOf((short) 0);
+            return (short) 0;
         } else if (clazz == Integer.TYPE) {
-            return Integer.valueOf(0);
+            return 0;
         } else if (clazz == Long.TYPE) {
-            return Long.valueOf(0);
+            return 0L;
         } else if (clazz == Float.TYPE) {
-            return Float.valueOf(0);
+            return (float) 0;
         } else if (clazz == Double.TYPE) {
-            return Double.valueOf(0);
+            return (double) 0;
```

Comment:
Why `0L` is used for `long`, but `0f` and `0d` aren't used for `float` and `double`?

Reply:
I'll fix it. Would you prefer a rebase/amend or a new commit and merges?